auth/src/main/java/org/openedx/auth/data/repository/AuthRepository.kt

```diff
@@ -12,6 +13,16 @@ class AuthRepository(
     private val preferencesManager: CorePreferences,
 ) {
 
+    private suspend fun processAuthResponse(authResponse: AuthResponse) {
+        if (authResponse.error != null) {
+            throw EdxError.UnknownException(authResponse.error!!)
+        }
+        preferencesManager.accessToken = authResponse.accessToken ?: ""
+        preferencesManager.refreshToken = authResponse.refreshToken ?: ""
+        val user = api.getProfile().mapToDomain()
+        preferencesManager.user = user
+    }
```

Comment:
Extracted out the common code since both login functions only have a slightly different API call but the same response.